src/librustdoc/passes/collect_trait_impls.rs

```diff
@@ -131,12 +131,8 @@ crate fn collect_trait_impls(krate: Crate, cx: &mut DocContext<'_>) -> Crate {
         }
     }
 
-    let items = if let Some(ref mut it) = krate.module {
-        if let ModuleItem(Module { ref mut items, .. }) = *it.kind {
-            items
-        } else {
-            panic!("collect-trait-impls can't run");
-        }
+    let items = if let ModuleItem(Module { ref mut items, .. }) = *krate.module.kind {
```

Comment:
Seems like a future enhancement could be to store a `Module` instead of an `Item`. Probably we would need to add a few more fields to `Module` (like DefId and name) to do that.